### DISCUSSION TITLE
publish ruff wasm to npm?

### DISCUSSION THREAD

**Discussion by zhanba** [astral-sh/ruff#4425] [Ideas]
- the ruff playground have used ruff wasm build, can we publish it to npm? it would be very useful for web based python editor

**Comment by dikwickley** [comment]
hi @zhanba can you point towards the ruff wasm bin?

**Reply by MichaReiser** [reply]
There's no binary, but you can find the source in `crates/ruff_wasm`. 

I can see how a `ruff_wasm` npm package would simplify usage but increases our release pipeline's complexity. You could try use git submodules and directly reference the `ruff_wasm` crate instead.

**Reply by dikwickley** [reply]
Thank you!

**Comment by JonathanPlasse** [comment]
- Would having ruff wasm publish to npm allow multi-version playground requested in #1385?

**Reply by MichaReiser** [reply]
It could help in the sense that we don't need to rebuild older versions when publishing the playground but it still requires hosting each version and committing to an API that is stable enough (or the playground must handle different versions)
